src/AutoRest.CSharp/Common/Output/Models/Requests/RestClientMethod.cs

```diff
@@ -16,51 +16,83 @@ public RestClientMethod(string name, string? description, CSharpType? returnType
         {
             Name = name;
             Request = request;
-            Parameters = parameters;
             Responses = responses;
             Description = description;
             ReturnType = returnType;
             HeaderModel = headerModel;
             BufferResponse = bufferResponse;
             Accessibility = accessibility;
             Operation = operation;
+
+            PathParameters = getPathParameters(parameters);
+            NonPathParameters = getNonPathParameters(parameters);
+
+            Parameters = SortParameters(parameters, new Queue<Parameter>(PathParameters));
         }
 
         public string Name { get; }
         public string? Description { get; }
         public Request Request { get; }
-        public Parameter[] Parameters { get; }
+        public List<Parameter> Parameters { get; }
         public Response[] Responses { get; }
         public DataPlaneResponseHeaderGroupType? HeaderModel { get; }
         public bool BufferResponse { get; }
         public CSharpType? ReturnType { get; }
         public string Accessibility { get; }
         public Operation Operation { get; }
 
-        public List<Parameter> NonPathParameters
+        public List<Parameter> PathParameters { get; }
+
+        public List<Parameter> NonPathParameters { get; }
+
+        private List<Parameter> getPathParameters(Parameter[] parameters)
         {
-            get
-            {
-                var pathSegments = this.Request.PathParameterSegments;
+            List<Parameter> pathParameters = new();
+            // Path segments are ordered from left to right, so iterate segments first will keep the sequence of
+            // path parameters in URL path
+            Request.PathParameterSegments.ForEach(pathSegment => {
+                var parameter = parameters.FirstOrDefault(p => pathSegment.Value.Reference.Type.Name == p.Type.Name &&
+                    pathSegment.Value.Reference.Name == p.Name);
+                if (parameter != null)
+                {
+                    pathParameters.Add(parameter);
+                }
+            });
 
-                return this.Parameters.Where(parameter => !pathSegments.Any(
-                    pathSegment => pathSegment.Value.Reference.Type.Name == parameter.Type.Name &&
-                    pathSegment.Value.Reference.Name == parameter.Name)
-                ).ToList();
-            }
+            return pathParameters;
```

Comment:
This looks the same as the condition `IsPathParameter`, therefore maybe we could just simplify this like this
```suggestion
return parameters.Where(p => IsPathParameter(p)).ToList();
```
so that this function could be coherent with `getNonPathParameters` to make sure that one parameter in the input make be in either this array or the other array.